lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -0,0 +1,510 @@
+  published: false,
+  duration: 10,
+  files: [
+    fakeFile(),
+  ],
+  error: {},
+  productVolume: 1119742,
+  timeToPreprocess: 0,
+  beginningDateTime: dateString,
+  endingDateTime: dateString,
+  processingStartDateTime: dateString,
+  processingEndDateTime: dateString,
+  lastUpdateDateTime: dateString,
+  timeToArchive: 0,
+  productionDateTime: dateString,
+  timestamp: Date.now(),
+  createdAt: Date.now() - 200 * 1000,
+  updatedAt: Date.now(),
+});
+
+let granulesModel;
+
+const testDbName = `data_migration_2_${cryptoRandomString({ length: 10 })}`;
+
+test.before(async (t) => {
+  await s3Utils.createBucket(bucket);
+  process.env.stackName = cryptoRandomString({ length: 10 });
+  process.env.system_bucket = bucket;
+
+  process.env.CollectionsTable = cryptoRandomString({ length: 10 });
+  process.env.ExecutionsTable = cryptoRandomString({ length: 10 });
+  process.env.GranulesTable = cryptoRandomString({ length: 10 });
+
+  granulesModel = new Granule();
+  await granulesModel.createTable();
+
+  t.context.granulePgModel = new GranulePgModel();
+  t.context.filePgModel = new FilePgModel();
+
+  const { knexAdmin, knex } = await generateLocalTestDb(
+    testDbName,
+    migrationDir
+  );
+  t.context.knexAdmin = knexAdmin;
+  t.context.knex = knex;
+});
+
+test.beforeEach(async (t) => {
+  const collectionPgModel = new CollectionPgModel();
+  const testCollection = fakeCollectionRecordFactory();
+
+  const collectionResponse = await collectionPgModel.create(
+    t.context.knex,
+    testCollection
+  );
+  t.context.testCollection = testCollection;
+  t.context.collectionCumulusId = collectionResponse[0];
+
+  const executionPgModel = new ExecutionPgModel();
+  const testExecution = fakeExecutionRecordFactory();
+
+  const executionResponse = await executionPgModel.create(
+    t.context.knex,
+    testExecution
+  );
+  t.context.testExecution = testExecution;
+  t.context.executionCumulusId = executionResponse[0];
+
+  t.context.testGranule = generateTestGranule(testCollection, testExecution.arn);
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.knex(tableNames.files).del();
+  await t.context.knex(tableNames.granules).del();
+  await t.context.knex(tableNames.collections).del();
+  await t.context.knex(tableNames.executions).del();
+});
+
+test.after.always(async (t) => {
+  await granulesModel.deleteTable();
+
+  await s3Utils.recursivelyDeleteS3Bucket(bucket);
+
+  await destroyLocalTestDb({
+    ...t.context,
+    testDbName,
+  });
+});
+
+test.serial('migrateGranuleRecord correctly migrates granule record', async (t) => {
+  const {
+    collectionCumulusId,
+    executionCumulusId,
+    granulePgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  await migrateGranuleRecord(testGranule, knex);
+
+  const record = await granulePgModel.get(knex, {
+    granule_id: testGranule.granuleId,
+    collection_cumulus_id: collectionCumulusId,
+  });
+
+  t.deepEqual(
+    omit(record, ['cumulus_id']),
```

Comment:
Since we could get the cumulus ID from `migrateGranuleRecord` above, we don't have to omit it and could do a direct comparison. But it's not required